File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        # IIV - invalid 
        """
        create a dictionary
        len of string
        iterate over string from right to left
        if value of left_character > value of right_character : add to sum
        else : subtract from sum
        """
        rom_int = {"I" : 1, "V" : 5, "X" : 10, "L" : 50, "C" : 100, 
                   "D" : 500, "M" : 1000}
        str_len = len(s)
        int_value = rom_int[s[str_len-1]]
        
        for i in range(str_len - 2, -1, -1):
            if rom_int[s[i]] >= rom_int[s[i+1]]:
                int_value += rom_int[s[i]]
            else:
                int_value -= rom_int[s[i]]
        
        return int_value
        

File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> romInt = new HashMap<Character, Integer>();
        romInt.put('I', 1);
        romInt.put('V', 5);
        romInt.put('X', 10);
        romInt.put('L', 50);
        romInt.put('C', 100);
        romInt.put('D', 500);
        romInt.put('M', 1000);
        
        int n = s.length();
        int intValue=0;
        
        for (int i=0; i<n-1; i++){
            if (romInt.get(s.charAt(i)) >= romInt.get(s.charAt(i+1))){
                intValue += romInt.get(s.charAt(i));
            } else{
                intValue -= romInt.get(s.charAt(i));
            }
        }
        intValue += romInt.get(s.charAt(n-1));
        return intValue;
    }
}